garak/generators/ggml.py

```diff
@@ -34,51 +36,81 @@ class GgmlGenerator(Generator):
     top_p = 0.95
     temperature = 0.8
     exception_on_failure = True
+    first_call = True
 
     generator_family_name = "ggml"
 
+    def command_params(self):
+        return {
+            "-m": self.name,
+            "-n": self.max_tokens,
+            "--repeat-penalty": self.repeat_penalty,
+            "--presence-penalty": self.presence_penalty,
+            "--frequency-penalty": self.frequency_penalty,
+            "--top-k": self.top_k,
+            "--top-p": self.top_p,
+            "--temp": self.temperature,
+            "-s": self.seed,
+        }
+
+
     def __init__(self, name, generations=10):
         self.path_to_ggml_main = os.getenv("GGML_MAIN_PATH")
+        if self.path_to_ggml_main is None:
+            raise RuntimeError("Executable not provided by environment GGML_MAIN_PATH")
+        if not os.path.isfile(self.path_to_ggml_main):
+            raise RuntimeError("Unable to locate executable")
```

Comment:
Maybe slightly more descriptive? 
`f"Unable to locate ggml executable at {self.path_to_ggml_main}"`

Reply:
Or maybe even tighter? 

`f"Looked for executable at '{self.path_to_ggml_main}' but this wasn't a file"`

Unable to locate implies there was an effort to locate, but we only check in one pre-specified location

Also - I'm split between RuntimeError and FileNotFoundError. The latter seems more specific though.

Reply:
I think `RuntimeError` is probably correct here? There is a path, and it may even exist -- but it's not a file (it could be a directory that contains the relevant file and you need to update `GGML_MAIN_PATH`!) IMO, `FileNotFoundError` is for when the path doesn't exist.

Either error message is probably fine so long as it communicates "We looked where your env points and it wasn't a file, so that's not the right executable path."

Reply:
"Raised when a file or directory is requested but doesn’t exist. Corresponds to errno [ENOENT](https://docs.python.org/3/library/errno.html#errno.ENOENT)." seemed to fit for me - one requests a file, it's not there. But hey, if this is the most contentious point in the PR, we're in a good place. And the exception definitely occurs at runtime.

Reply:
Yeah, I'm with you on that. I that line 60 should definitely be a `FileNotFoundError` but I'm not willing to die on whether 62 should be `RuntimeError` or `FileNotFoundError`.

Reply:
Good choice. There are much better maxima out there.

Reply:
Interesting points, line 60 should stay `RuntimeError` as that is simply testing if the env variable was set at all it the test there did not attempt to access the filesystem in any way to warrant `FileNotFoundError`. This is a runtime based value, I might be convinced it should be `ValueError` however that seems a stretch.

> ValueError is raised when a function receives an argument of the correct type but an inappropriate value.

Line 62 can be more specific as `FileNotFoundError` as the code actually looked for the file. Scoping the message better is a great call out.

I will also update line 69 as `FileNotFoundError` and adjust the message reported.